src/OrleansRuntime/Catalog/GrainLifecycle.cs

```diff
@@ -0,0 +1,10 @@
+﻿
+namespace Orleans.Runtime
+{
+    internal class GrainLifecycle : LifecycleObservable<GrainLifecyleStage>, IGrainLifeCycle
```

Comment:
For what it currently does, yes, this is a little heavy.  But the plan is to use the lifecycle to manage the full grain lifecycle, not just the facets.